### PR TITLE
don't cross up aspect events with primary build events

### PR DESCRIPTION
**Pull request by jdhollen**
Also, set things up so that we _could_ track aspects if we ever wanted to..

Fixes https://github.com/buildbuddy-io/buildbuddy-internal/issues/3603

**Related issues**: N/A

### CODE REVIEW

File: server/build_event_protocol/target_tracker/target_tracker.go [hunk 1]
```diff
@@ -112,15 +114,17 @@ func (t *target) updateFromEvent(event *build_event_stream.BuildEvent) {
 	}
 }
 
-func getLabelFromEventId(beid *build_event_stream.BuildEventId) string {
+func getTargetIdWithAspectFromEventId(beid *build_event_stream.BuildEventId) string {
 	switch beid.Id.(type) {
 	case *build_event_stream.BuildEventId_TargetConfigured:
-		return beid.GetTargetConfigured().GetLabel()
+		return beid.GetTargetConfigured().GetLabel() + beid.GetTargetConfigured().GetAspect()
```
[the remark this answers is not in the document]
Comment (jdhollen) [reply server/build_event_protocol/target_tracker/target_tracker.go]: Sure sure, done

File: server/build_event_protocol/target_tracker/target_tracker.go [hunk 2]
```diff
@@ -112,15 +114,27 @@ func (t *target) updateFromEvent(event *build_event_stream.BuildEvent) {
 	}
 }
 
-func getLabelFromEventId(beid *build_event_stream.BuildEventId) string {
+const target_id_separator string = "|"
```
Comment (fmeum) [inline server/build_event_protocol/target_tracker/target_tracker.go:117]: In Bazel, for example when specifying an aspect on the command line, the separator is usually `%`.

Comment (jdhollen) [reply server/build_event_protocol/target_tracker/target_tracker.go:117]: This is just a separator for generating an id out of a [target, aspect?] tuple--the aspect string is formatted as you describe, so it comes out as `//some/regular:target|//some/random.bzl%aspect`

**Review by luluz66 — APPROVED** [review]

File: server/build_event_protocol/target_tracker/target_tracker.go (hunk 2 above)
Comment (luluz66) [reply server/build_event_protocol/target_tracker/target_tracker.go:117]: nit: Camel case?

**Pull request merged**